pydantic_ai_slim/pydantic_ai/settings.py

```diff
@@ -0,0 +1,30 @@
+from __future__ import annotations
+
+from httpx import Timeout
+from typing_extensions import TypedDict
+
+
+class ModelSettings(TypedDict, total=False):
+    """The runtime settings for a model's request / message creation."""
+
+    max_tokens: int
+    """The maximum number of tokens to generate before stopping."""
```

Comment:
maybe for each setting, we should state which of our supported models respect them?